src/Assert.php

```diff
@@ -1582,6 +1597,8 @@ public static function lengthBetween($value, $min, $max, $message = '')
     /**
      * Will also pass if $value is a directory, use Assert::file() instead if you need to be sure it is a file.
      *
+     * @psalm-pure
```

Comment:
absolutely not pure: this call depends on global state (filesystem), and also executes in non-deterministic time